src/components/Accordion/Accordion.js

```diff
@@ -44,28 +55,43 @@ class Accordion extends Component {
 
   constructor(props) {
     super(props);
-    const { open } = props;
-    if (open === null) {
-      this.state = { activeSection: props.defaultOpen ? props.defaultOpen : props.children[0].props.id };
+    const { children, defaultOpen, open } = props;
+    if (open === null) { // controlled
+      if (defaultOpen) {
+        if (typeof defaultOpen === 'string') {
+          this.state = { activeSections: [defaultOpen] };
+        } else { // is array of strings
+          this.state = { activeSections: defaultOpen };
+        }
+      } else {
+        this.state = { activeSections: [children[0].props.id] };
+      }
     }
   }
 
   handleClick = (id) => {
-    const { open, onSectionClick } = this.props;
-    if (open === null) {
-      this.setState({ activeSection: id });
-    } else {
-      onSectionClick(id);
+    const { open, onSectionClick, multiple } = this.props;
+    if (open !== null) { // controlled
+      onSectionClick(id); return;
     }
+    // uncontrolled
+    this.setState((prevState) => {
```

Comment:
Naming this callback could reduce complexity here, e.g: 

``` js
const toggleSectionInState = (...) => {};

this.setState(toggleSectionInState);
```

Reply:
after destructuring the prevState, it looks a lot simpler